tests/functional/aws-node-sdk/test/multipleBackend/objectTagging.js

```diff
@@ -0,0 +1,264 @@
+                    s3.putObject(params, err => {
+                        assert.equal(err, null);
+                        getObject(backend, true, false, done);
+                    });
+                });
+
+                it(`should put a 0 byte object with tags to ${backend} backend`,
+                done => {
+                    const params = Object.assign({ Tagging: tagString,
+                        Metadata: { 'scal-location-constraint': backend } },
+                        noBodyParams);
+                    s3.putObject(params, err => {
+                        assert.equal(err, null);
+                        getObject(backend, true, true, done);
+                    });
+                });
+
+                it(`should put tags to preexisting object in ${backend} ` +
+                'backend', done => {
+                    const params = Object.assign({ Metadata:
+                        { 'scal-location-constraint': backend } }, putParams);
+                    s3.putObject(params, err => {
+                        assert.equal(err, null);
+                        const putTagParams = { Bucket: bucket, Key: key,
+                            Tagging: putTags };
+                        s3.putObjectTagging(putTagParams, err => {
+                            assert.equal(err, null);
+                            getObject(backend, true, false, done);
+                        });
+                    });
+                });
+
+                it('should put tags to preexisting 0 byte object in ' +
+                `${backend} backend`, done => {
+                    const params = Object.assign({ Metadata:
+                        { 'scal-location-constraint': backend } },
+                        noBodyParams);
+                    s3.putObject(params, err => {
+                        assert.equal(err, null);
+                        const putTagParams = { Bucket: bucket, Key: key,
+                            Tagging: putTags };
+                        s3.putObjectTagging(putTagParams, err => {
+                            assert.equal(err, null);
+                            getObject(backend, true, true, done);
+                        });
+                    });
+                });
+            });
+
+            it('should return error on putting tags to object deleted from AWS',
+            done => {
+                const params = Object.assign({ Metadata:
+                    { 'scal-location-constraint': awsLocation } }, putParams);
+                s3.putObject(params, err => {
+                    assert.equal(err, null);
+                    awsS3.deleteObject(awsDelParams, err => {
+                        assert.equal(err, null);
+                        const putTagParams = { Bucket: bucket, Key: key,
+                            Tagging: putTags };
+                        s3.putObjectTagging(putTagParams, err => {
+                            assert.strictEqual(err.code, 'InternalError');
```

Comment:
`InternalError` sounds a bit too generic and scary for this case, and it does not give back any useful info to the user, maybe simply ObjNotFound would be more appropriate? Or maybe we need to create new specific errors when there are inconsistencies caused by the user accessing AWS directly. I understand that we're talking about the object being removed from AWS but not from scality S3 backend.